source/geometry/Quaternion.ooc

```diff
@@ -210,7 +210,7 @@ Quaternion: cover {
 	operator - -> This { This new(-this real, -this imaginary) }
 	operator / (value: Float) -> This { This new(this w / value, this x / value, this y / value, this z / value) }
 	operator * (value: Float) -> This { This new(this w * value, this x * value, this y * value, this z * value) }
-	operator * (value: FloatPoint3D) -> FloatPoint3D { (this * This new(0.0f, value) * this inverse) imaginary }
+	operator * (value: FloatPoint3D) -> FloatPoint3D { This hamiltonProduct(This hamiltonProduct(this, This new(0.0f, value)), this inverse) imaginary }
```

Comment:
No optimization at first. Tried now with different levels and seem to get approximately the same ratio for all of them.